clients/spark/core/src/main/scala/io/treeverse/gc/DataLister.scala

```diff
@@ -0,0 +1,25 @@
+package io.treeverse.gc
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.SparkSession
+import scala.collection.mutable.ListBuffer
+
+/** List all the files under a given path.
+ */
+trait DataLister {
+  def listData(spark: SparkSession, path: Path): DataFrame
+}
+
+class NaiveDataLister extends DataLister {
+  override def listData(spark: SparkSession, path: Path): DataFrame = {
+    import spark.implicits._
+    val fs = path.getFileSystem(spark.sparkContext.hadoopConfiguration)
+    val dataIt = fs.listFiles(path, false)
```

Comment:
We need an exclusion list. For example `dummy` or anything under `_lakefs` should not be listed.

Reply:
done